lib/puppet_x/bodeco/util.rb

```diff
@@ -42,8 +42,13 @@ def generate_request(uri)
       end
 
       def follow_redirect(uri, option = { limit: FOLLOW_LIMIT }, &block)
-        Net::HTTP.start(uri.host, uri.port, use_ssl: (uri.scheme == 'https')) do |http|
-          http.verify_mode = OpenSSL::SSL::VERIFY_NONE if @insecure
```

Comment:
`Net::HTTP.start` fails due to certificate verify error and the code in the block (where we attempt to ignore cert errors) is never executed.